llvm/lib/CodeGen/AtomicExpandPass.cpp

```diff
@@ -1988,7 +1988,7 @@ bool AtomicExpand::expandAtomicOpToLibcall(
   std::string LibcallName = TLI->getLibcallName(RTLibType);
   // We are compiling for CHERI purecap mode if the default globals address
   // space is a capability type.
-  bool IsCheriPurecap = DL.isFatPointer(DL.getGlobalsAddressSpace());
+  bool IsCheriPurecap = DL.isFatPointer(DL.getDefaultGlobalsAddressSpace());
```

Comment:
This one should probably be using the targetinfo instead of checking indirectly but that can be fixed separately.

Reply:
Thanks. I've included a patch to use the ABI info to deduce purecap.